profiler/shakesapp/shakesapp_main.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Shakesapp is a web application which starts up a server that can be
+// queried to determine how many times a string appears in the works of
+// Shakespeare and then sends requests to that server.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"time"
+
+	"cloud.google.com/go/profiler"
+	"google.golang.org/grpc"
+
+	"shakesapp/shakesapp"
+)
+
+var (
+	version      = flag.String("version", "1.0.0", "application version")
+	projectID    = flag.String("project_id", "", "project ID to run profiler with; required when running outside of GCP.")
```

Comment:
"required" -> "only required"

Reply:
or "required when" -> "required when and only when"

Reply:
Done.